strapi/strapi
# feat(content-history): add version header

### What does it do?

- Refactors to flatten VersionDetails into separated components: Header and Content 
- Add the version.createAt date and time as the header title
- Add the contentType's 'mainField' (entry title) value from the version data, if it does not exist fallback to the id
- Add locale object to the version data response to return both the locale code and name

For the locale:

The response for a version returns a `locale` property that is either a string or null. Currently proposes changing the response to return a `locale` property that is either an object `{name: string, code: string}` or null. However, this requires a check for the i18n plugin.

I know it was mentioned that we should inject a component from i18n. But I'm hesitant to do that here because it means creating an [injection zone](https://docs.strapi.io/dev-docs/migration/v3-to-v4/plugin/migrate-front-end#adding-injection-zones) or [creating a hook](https://docs.strapi.io/dev-docs/api/plugins/admin-panel-api#hooks-api) in content-history which I think would allow any plugin to inject components or mutate the header of the content-history page and I'm not sure that's what we want...

### Why is it needed?

- To display basic information about a history version

### How to test it?

- Open a history version that does NOT have i18n. You should see something like:
![Screenshot 2024-02-08 at 15 39 01](https://github.com/strapi/strapi/assets/26598053/d4417b37-3be2-4b2a-a546-556b4f3892fd)
- Open a history version that does have i18n: You should see something like:
![Screenshot 2024-02-08 at 15 40 11](https://github.com/strapi/strapi/assets/26598053/6e8064ae-4e30-4ac8-9958-d9cc93781d4d)
- Change the admin language in `/admin/me`, go back to a history version. You should see the date time format has changed:
![Screenshot 2024-02-08 at 15 40 55](https://github.com/strapi/strapi/assets/26598053/74579699-ecfd-4687-9519-694d66522ea3)
- Uninstalling i18n breaks the content-manager on v5/main so nothing to test there.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
